tests/auth_tests/test_mixins.py

```diff
@@ -0,0 +1,121 @@
+from django.conf import settings
+from django.contrib.auth import models
+from django.contrib.auth.mixins import (
+    LoginRequiredMixin, PermissionRequiredMixin,
+)
+from django.contrib.auth.models import AnonymousUser
+from django.core.exceptions import PermissionDenied
+from django.http import HttpResponse
+from django.test import RequestFactory, TestCase, override_settings
+from django.views.generic import View
+
+from .test_views import AuthViewsTestCase
+
+
+@override_settings(ROOT_URLCONF='auth_tests.urls')
+class LoginRequiredMixinTestCase(AuthViewsTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = models.User.objects.create(username='joe', password='qwerty')
+        cls.factory = RequestFactory()
+
+    def test_login_required(self, view=None, login_url=None):
+        """
+        Check that login_required works on a simple view wrapped in a
+        login_required decorator.
+        """
+        if login_url is None:
+            login_url = settings.LOGIN_URL
+
+        if view is None:
+            class AView(LoginRequiredMixin, View):
+
+                def get(self, request, *args, **kwargs):
+                    return HttpResponse()
+
+            view = AView.as_view()
+
+        request = self.factory.get('/rand')
+        request.user = AnonymousUser()
+        response = view(request)
+        self.assertEqual(response.status_code, 302)
+        self.assertIn(login_url, response.url)
```

Comment:
Yes, but apparently it only works with the test client, not with RequestFactory. I'll change the assertIn to an equality check, but I don't think it is worth creating a bunch of URLs for the test views in the URL config just to test these mixins.